ui/src/components/SearchInput/index.vue

```diff
@@ -0,0 +1,221 @@
+  }
+};
+
+const handleCloseSearch = () => closeSearch(true);
+
+const positionRight = computed(() => {
+  const w = width.value;
+
+  if (!drawerRef.value || w === 0) return `${gutterWidth}px`;
+
+  const clamped = Math.min(800, Math.max(600, Math.round(w)));
+  return `calc(${clamped}px + ${gutterWidth}px)`;
+});
+
+watch(
+  () => searchKey.value,
+  value => {
+    if (!value) {
+      props.searchAddon.clearDecorations();
+      props.searchAddon.clearActiveDecoration();
+    }
+  }
+);
+
+onMounted(() => {
+  drawerRef.value = document.getElementById('drawer-inner-target');
+});
+
+useMutationObserver(
+  () => document.body,
+  () => {
+    const el = document.getElementById('drawer-inner-target');
+
+    if (el !== drawerRef.value) {
+      drawerRef.value = el;
+    }
+  },
+  {
+    childList: true,
+    subtree: true,
+  }
+);
+</script>
+
+<template>
+  <div
+    class="absolute flex items-center right-2 p-2 z-999 shadow-md rounded-md"
+    :style="{ backgroundColor: lighten(10), top: isKubernetes ? '3rem' : '0.5rem', right: positionRight }"
+  >
+    <n-flex align="center" class="!gap-2">
+      <n-input
+        v-model:value="searchKey"
+        clearable
+        size="small"
+        :style="{ width: '45%', flex: '1' }"
+        @update:value="keyWordsSearch"
+      />
+
+      <n-button quaternary size="tiny" @click="handleSearchPrevious">
+        <template #icon>
+          <ChevronUp />
+        </template>
+      </n-button>
+      <n-button quaternary size="tiny" @click="handleSearchNext">
+        <template #icon>
+          <ChevronDown />
+        </template>
+      </n-button>
+    </n-flex>
+
+    <n-divider vertical :style="{ height: '24px', backgroundColor: darken(1) }" />
+
+    <n-flex align="center" class="!flex-nowrap !gap-2">
+      <n-flex align="center" class="!flex-nowrap !gap-2">
+        <div
+          v-for="option in searchOptionButtons"
+          :key="option.key"
+          class="search-option-btn"
+          :class="{ active: searchOptions[option.key] }"
+          :style="getActiveStyle(Boolean(searchOptions[option.key]))"
+          :title="option.label"
+          tabindex="0"
+          @click="option.toggleFn"
+        >
+          <component :is="option.icon" color="#fff" :size="16" />
+        </div>
+      </n-flex>
+
+      <n-button quaternary size="tiny" @click="handleCloseSearch">
+        <template #icon>
+          <X />
+        </template>
+      </n-button>
+    </n-flex>
+  </div>
+</template>
+
+<style scoped lang="scss">
+.search-option-btn {
+  padding: 4px 6px;
+  border-radius: 4px;
+  cursor: pointer;
+  transition: all 0.2s ease-in-out;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  user-select: none;
+  background-color: transparent;
+
+  &:hover {
+    background-color: rgba(255, 255, 255, 0.1);
+    transform: scale(1.05);
+  }
+
+  &:active {
+    background-color: rgba(255, 255, 255, 0.2);
+    transform: scale(0.95);
+  }
+}
+</style>
```

Comment:
The code seems to be well written within the constraints provided. However, no specific issues or optimization suggestions were mentioned. It looks like there might be some formatting inconsistencies which could potentially improve readability. If you have particular concerns regarding this file please let me know how I can help further with details about these points.